Makefile

```diff
@@ -260,7 +265,8 @@ endif
 	docker run --rm -v $(CURDIR)/dist/bin:/go/bin:rw $(CALICO_BUILD) /bin/sh -c "\
 	  echo; echo calico-node-$(ARCH) -v;	 /go/bin/calico-node-$(ARCH) -v; \
 	"
-	docker build --pull -t $(NODE_IMAGE):latest-$(ARCH) . --build-arg BIRD_IMAGE=$(BIRD_IMAGE) --build-arg QEMU_IMAGE=$(CALICO_BUILD) --build-arg GIT_VERSION=$(GIT_VERSION) -f ./Dockerfile.$(ARCH)
+## --platform fixes an issue where FROM SCRATCH imports an amd64 layer causing an architecture problem in cross compiling.
```

Comment:
Is there some sort of upstream bug documenting this? Or is this intended behavior and our scripts were just wrong?

Reply:
Without a target platform `FROM SCRATCH` imports a amd64 image in a cross compilation run, this results in registeries tagging our image as amd64 instead of arm.

https://github.com/projectcalico/node/pull/1044#discussion_r660922983

Reply:
@caseydavenport I reckon there was no documentation stating `scratch` situation, the only thing that I was able to found in terms of documentation was the link  in my previous comment. Reading this official docker blog was also an indication that my solution is sound. https://www.docker.com/blog/multi-platform-docker-builds/#:~:text=If%20we%20build%20it%20normally%20and%20run%20it%3A